### PR TITLE
Sprint6 datacleanup to development branch

### PR DESCRIPTION
**Pull request by vvaishu05**

### CODE REVIEW

File: crud/clicks_resample.py [hunk 1]
```diff
@@ -134,13 +134,32 @@ def update_click(db:session, click_count:int):
...
+	Args:
+		db (session): DB connection session for ORM functionalities
+		delete_date (str, optional): Date-time. Defaults to =dt.strftime(dt.utcnow(), "%Y-%m-%dT%H:%M:%S").
+
+	Returns:
+		list: deleted click records
+	"""
+	try:
+		# Query data to be deleted for archiving
+		to_delete_data = db.query(ClicksResample).filter(ClicksResample.click_sampled_timestamp<delete_date).all()
```
Comment (uditsrn28) [inline crud/clicks_resample.py:155]: Remove this

File: crud/clicks_resample.py [hunk 2]
```diff
@@ -134,13 +134,32 @@ def update_click(db:session, click_count:int):
...
+		print("Resampled clicks to be deleted", len(to_delete_data))
```
Comment (uditsrn28) [inline crud/clicks_resample.py:156]: Remove this or convert to log

File: crud/clicks_resample.py [hunk 3]
```diff
@@ -134,13 +134,32 @@ def update_click(db:session, click_count:int):
...
+		# If data is present for the given time range, delete them
+		if len(to_delete_data) > 0:
+			deleted = db.query(ClicksResample).filter(ClicksResample.click_sampled_timestamp<delete_date).delete()
+			print("Deleted sampleclicks", deleted)
```
Comment (uditsrn28) [inline crud/clicks_resample.py:160]: Remove this or convert to log

File: crud/views.py [hunk 4]
```diff
@@ -113,3 +113,24 @@ async def delete_all_views():
...
+		older_than_date (str, optional): Date-time. Defaults to dt.strftime(dt.utcnow(), "%Y-%m-%dT%H:%M:%S").
+
+	Returns:
+		list: deleted view records
+	"""
+	mongoDBConnection = mongoDB.database
+	# Query data to be deleted for archiving
+	cursor = mongoDBConnection["views"].find().where(f"this.view_created < '{older_than_date}'").sort("view_created", ASCENDING)
+	to_delete_data = await cursor.to_list(length=10000)
+	print("Rows to be deleted", len(to_delete_data))
```
Comment (uditsrn28) [inline crud/views.py:131]: Remove this or convert to log

File: crud/views.py [hunk 5]
```diff
@@ -113,3 +113,24 @@ async def delete_all_views():
...
+	# If data is present for the given time range, delete them
+	if len(to_delete_data) > 0:
+		deleted_data = await mongoDBConnection["views"].delete_many({"view_created": {"$lte": all[-1]["view_created"]}})
+		print("Deleted views >> ", deleted_data.deleted_count)
```
Comment (uditsrn28) [inline crud/views.py:135]: Remove this or convert to log

File: crud/views_resample.py [hunk 6]
```diff
@@ -134,16 +146,35 @@ def update_view(db:session, view_id:int, view_count:int=None, device_name:str=No
...
+		db (session): DB connection session for ORM functionalities
+		delete_date (str, optional): Date-time. Defaults to =dt.strftime(dt.utcnow(), "%Y-%m-%dT%H:%M:%S").
+
+	Returns:
+		list: deleted view records
+	"""
+	try:
+		# Query data to be deleted for archiving
+		to_delete_data = db.query(ViewsResample).filter(ViewsResample.view_sampled_timestamp<delete_date).all()
+		print("Resampled views to be deleted", len(to_delete_data))
```
Comment (uditsrn28) [inline crud/views_resample.py:171]: Remove this or convert to log

File: crud/views_resample.py [hunk 7]
```diff
@@ -134,16 +146,35 @@ def update_view(db:session, view_id:int, view_count:int=None, device_name:str=No
...
+		# If data is present for the given time range, delete them
+		if len(to_delete_data) > 0:
+			deleted_data = db.query(ViewsResample).filter(ViewsResample.view_sampled_timestamp<delete_date).delete()
+			print("Deleted sampleviews", deleted_data)
```
Comment (uditsrn28) [inline crud/views_resample.py:175]: Remove this or convert to log

File: router/analytics.py [hunk 8]
```diff
@@ -224,4 +224,47 @@ async def get_location_activity(username: str, start_date: dt, end_date: dt, db:
...
+		click_count = click_count.mappings().all()
+		if len(click_count) == 0:
+			return JSONResponse(content={"message": "Data not found for the given date range"}, status_code=status.HTTP_404_NOT_FOUND)
+		
+		click_count_df = pd.DataFrame(click_count)
+		# To get the total clicks count for each link
+		clicks_grouped_by_link = click_count_df.groupby([click_count_df.link_name])["click_count"].sum()
+		# TODO: To try sorting by counts in descending order
+		# clicks_grouped_by_link.sort_values('click_count', ascending=False, inplace=True)
+		print(clicks_grouped_by_link)
```
Comment (uditsrn28) [inline router/analytics.py:263]: Remove this or convert to log

File: tasks/task.py [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+	# As FastAPI isnt used directly, SQL DB session needs to be opened/closed separately
+	db:session=next(get_db())
+	all_profiles = profiles.get_all_profiles(db)
+	# Sample views and clicks for each profile separately
+	for profile in all_profiles:
+		last_queried = profile.profile_description
+		if last_queried not in ['', "", None, "''"]:
+			last_queried = json.loads(last_queried)
+  
+		print(f"profile_last_queried for profile {profile.id} ++", last_queried)
```
Comment (uditsrn28) [inline tasks/task.py:52]: Remove this or convert to log

File: tasks/task.py [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+		# Each profile runs as a separate task
+		if is_celery_working() == True:
+			print("Celery working")
```
Comment (uditsrn28) [inline tasks/task.py:55]: Remove this or convert to log

File: tasks/task.py [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+			task = query_sample_view_clicks.delay(profile, last_queried)
+		else:
+			print("Celery not working")
```
Comment (uditsrn28) [inline tasks/task.py:58]: Remove this or convert to log

File: utilities/generic.py [hunk 12]
```diff
@@ -35,3 +40,145 @@ def valid_ip_address(addr: str) -> str:
...
 		return "Invalid"
 
+
+def removeFromPath(path:str):
+	"""Function to remove give file/folder path recursively
+
+	Args:
+		path (str): path which should be deleted
+	"""
+	print('removing ', path)
```
Comment (uditsrn28) [inline utilities/generic.py:50]: Remove this or convert to log

File: utilities/generic.py [hunk 13]
```diff
@@ -35,3 +40,145 @@ def valid_ip_address(addr: str) -> str:
...
+	try:
+		if os.path.isdir(path):
+			if not shutil.rmtree(path):
+				print(path, ' is removed successfully')
```
Comment (uditsrn28) [inline utilities/generic.py:54]: Remove this or convert to log

File: utilities/generic.py [hunk 14]
```diff
@@ -35,3 +40,145 @@ def valid_ip_address(addr: str) -> str:
...
+		else:
+			if not os.remove(path):
+				print(path, ' is removed successfully')
+	except Exception as e:
```
Comment (uditsrn28) [inline utilities/generic.py:58]: Remove this or convert to log

**Comment by uditsrn28** [comment]
Please remove print statements or convert them into log

File: utilities/generic.py (hunk 14 above)
Comment (vvaishu05) [reply utilities/generic.py]: Replaced with log

File: utilities/generic.py (hunk 13 above)
Comment (vvaishu05) [reply utilities/generic.py]: Replaced with log

File: utilities/generic.py (hunk 12 above)
Comment (vvaishu05) [reply utilities/generic.py]: Replaced with log

File: tasks/task.py (hunk 11 above)
Comment (vvaishu05) [reply tasks/task.py]: Replaced with log

File: tasks/task.py (hunk 9 above)
Comment (vvaishu05) [reply tasks/task.py]: Replaced with log

File: tasks/task.py (hunk 10 above)
Comment (vvaishu05) [reply tasks/task.py]: Replaced with log

File: router/analytics.py (hunk 8 above)
Comment (vvaishu05) [reply router/analytics.py]: Removed

File: crud/views_resample.py (hunk 7 above)
Comment (vvaishu05) [reply crud/views_resample.py]: Replaced with log

File: crud/views_resample.py (hunk 6 above)
Comment (vvaishu05) [reply crud/views_resample.py]: Replaced with log

File: crud/views.py (hunk 5 above)
Comment (vvaishu05) [reply crud/views.py]: Replaced with log

File: crud/views.py (hunk 4 above)
Comment (vvaishu05) [reply crud/views.py]: Replaced with log

File: crud/clicks_resample.py (hunk 3 above)
Comment (vvaishu05) [reply crud/clicks_resample.py]: Replaced with log

File: crud/clicks_resample.py (hunk 2 above)
Comment (vvaishu05) [reply crud/clicks_resample.py]: Replaced with log

**Comment by vvaishu05** [comment]
Squash-merged after resolving conflicts through terminal

**Pull request closed without merging**